docs/book/src/plugins/external-plugins.md

```diff
@@ -0,0 +1,168 @@
+# Extending Kubebuilder with external plugins
+
+## Overview
+
+Kubebuilder's functionality can be extended through the use of external plugins.
+
+An external plugin is an executable that implements an execution pattern that Kubebuilder knows how to interact with. Since an external plugin is an executable it can be implemented in any language. 
+
+The Kubebuilder CLI loads the external plugin in the specified path and interact the plugin through the executable's `stdin` & `stdout`.
+
+## When is it useful?
+
+- If you want to create helpers or addons on top of the scaffolds done by Kubebuilder's default scaffolding.
+
+- If you design customized layouts and want to take advantage of functions from Kubebuilder library.
+
+- If you are looking for implementing plugins in a language other than `Go`.
+
+## How to write it?
+
+The inter-process communication between your external plugin and Kubebuilder is through the standard I/O.
+
+You can write your external plugin in any language as long as it implements the `PluginRequest` and `PluginResponse` interfaces. 
```

Comment:
Nit: "interfaces" suggests a looser construct in programming ("interfaces" can be anything behind the scenes as long as they implement certain functionality). I would recommend using "types" as it is more strict and aligns with what the external plugin communication process in Kubebuilder expects. The response must be exactly the `PluginResponse` type and shouldn't have any additional fields

Reply:
Thank you for the nice explanation! 👍
I tried to reorganized this paragraph in the latest commit.